src/statement.d

```diff
@@ -5149,6 +5154,19 @@ public:
                 }
             }
         }
+
+        if (sc.func)
+        {
+            flags |= sc.func.flags & (FUNCFLAGcppExceptions | FUNCFLAGdExceptions);
+            sc.func.flags |= flags;
+            if (flags == (FUNCFLAGcppExceptions | FUNCFLAGdExceptions))
+            {
+                error("cannot mix catching D and C++ exceptions in the same function");
```

Comment:
In C++ the TType value is a pointer to some rtti data.  While I can accept being wrong, I always considered it to be:

- Found candidate (D/C++/other), get_ttype_entry_for it.
- If generic exception we are propagating is domestic (D exception class), proceed as usual.
- If both TType entry language and exception class hint at being C++, get the underlying exception type (cast exception to __cxa_exception*) and do pointer comparison with the TType value.